From 36d654a2fe1d0d624eb49917299bb4707c997936 Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Mon, 6 Jun 2022 17:35:18 -0400
Subject: [PATCH 1/4] MdeModulePkg/SataControllerDxe: Add support for drives in
  RAID mode

This resolves the problem of using drivers connected to Intel G33
builtin SATA controller when run from DuetPkg when the only configurable
modes in the BIOS are IDE and RAID.

Signed-off-by: Vitaly Cheptsov <vit9696@protonmail.com>
---
 MdeModulePkg/Bus/Pci/SataControllerDxe/SataController.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/MdeModulePkg/Bus/Pci/SataControllerDxe/SataController.c b/MdeModulePkg/Bus/Pci/SataControllerDxe/SataController.c
index f661efaec7..bb803902b3 100644
--- a/MdeModulePkg/Bus/Pci/SataControllerDxe/SataController.c
+++ b/MdeModulePkg/Bus/Pci/SataControllerDxe/SataController.c
@@ -321,7 +321,7 @@ SataControllerSupported (
     return EFI_UNSUPPORTED;
   }
 
-  if (IS_PCI_IDE (&PciData) || IS_PCI_SATADPA (&PciData)) {
+  if (IS_PCI_IDE (&PciData) || IS_PCI_SATADPA (&PciData) || IS_PCI_RAID (&PciData)) {
     return EFI_SUCCESS;
   }
 
@@ -462,7 +462,7 @@ SataControllerStart (
   if (IS_PCI_IDE (&PciData)) {
     Private->IdeInit.ChannelCount = IDE_MAX_CHANNEL;
     Private->DeviceCount          = IDE_MAX_DEVICES;
-  } else if (IS_PCI_SATADPA (&PciData)) {
+  } else if (IS_PCI_SATADPA (&PciData) || IS_PCI_RAID (&PciData)) {
     //
     // Read Ports Implemented(PI) to calculate max port number (0 based).
     //
-- 
2.36.1

